File: benchmark/c/cpw/count_factorial_numbers_in_a_given_range/count_factorial_numbers_in_a_given_range.c


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>



int count_factorial_numbers_in_a_given_range ( int low, int high ) ;
int count_factorial_numbers_in_a_given_range ( int low, int high ) {
  int fact = 1, x = 1;
  while ( fact < low ) {
    fact = fact * x;
    x ++;
  }
  int res = 0;
  while ( fact <= high ) {
    res ++;
    fact = fact * x;
    x ++;
  }
  return res;
}

File: translation_result/else_bms/fail/c/cpw/count_factorial_numbers_in_a_given_range/gemini/CAPR/count_factorial_numbers_in_a_given_range.rs

fn count_factorial_numbers_in_a_given_range(low: i32, high: i32) -> i32 {
    let mut fact = 1;
    let mut x = 1;
    while fact < low {
        fact *= x;
        x += 1;
    }
    let mut res = 0;
    while fact <= high {
        res += 1;
        fact *= x;
        x += 1;
    }
    res
}
